optaplanner-examples/src/main/java/org/optaplanner/examples/common/business/SolutionBusiness.java

```diff
@@ -359,8 +358,7 @@ public void doProblemChange(ProblemChange<Solution_> problemChange) {
         if (solver.isSolving()) {
             solver.addProblemChange(problemChange);
         } else {
-            problemChange.doChange(guiScoreDirector.getWorkingSolution(), problemChangeDirector);
-            guiScoreDirector.calculateScore();
+            problemChangeDirector.doProblemChange(problemChange);
```

Comment:
I wonder why is the freshly calculated score not used...

Reply:
The `SolutionBusiness` has a reference to `guiScoreDirector`, which holds the working solution. The Swing frame refreshes the score by calling the `SolutionBusiness#getScore()`